pandas/_libs/lib.pyx

```diff
@@ -485,8 +485,7 @@ def astype_intsafe(ndarray[object] arr, new_dtype):
         bint is_datelike
         ndarray result
 
-    # on 32-bit, 1.6.2 numpy M8[ns] is a subdtype of integer, which is weird
-    is_datelike = new_dtype in ['M8[ns]', 'm8[ns]']
+    is_datelike = new_dtype == 'm8[ns]'
 
```

Comment:
core.dtypes.cast is the only place where this gets called, and it has a comment `# work around for NumPy brokenness, #1987`, might have more things worth cleaning up

Reply:
Removed this and the call in `dtypes.cast`. `pandas/tests/dtypes/test_cast.py` still passes, so let's try how this goes here.